weave/trace_server/calls_query_builder.py

```diff
@@ -1267,6 +1277,105 @@ def _create_like_optimized_in_condition(
     return "(" + " OR ".join(like_conditions) + ")"
 
 
+def _uuidv7_from_timestamp_zeroed(ms_since_epoch: int) -> str:
+    if not (0 <= ms_since_epoch < 2**48):
+        raise ValueError("Timestamp must be a 48-bit integer (ms since epoch).")
+
+    # Split the 48-bit timestamp
+    time_low = (ms_since_epoch >> 16) & 0xFFFFFFFF  # First 32 bits
+    time_mid = ms_since_epoch & 0xFFFF  # Next 16 bits
+
+    # Set version (7) in high 4 bits, rest zero
+    time_hi_and_version = 0x7 << 12  # Version 7 + 12 zero bits
+
+    # Set variant bits (10xx....) and rest zero
+    clock_seq_hi_and_reserved = 0x80  # 10xx xxxx
+    clock_seq_low = 0x00
+    node = 0x000000000000  # 48 bits of zero
+
+    # Create UUID from fields
+    uuid_fields = (
+        time_low,
+        time_mid,
+        time_hi_and_version,
+        clock_seq_hi_and_reserved,
+        clock_seq_low,
+        node,
+    )
+    return str(uuid.UUID(fields=uuid_fields))
+
+
+def _create_datetime_optimization_sql(
```

Comment:
I think the main limitations are:
1. This will be invalid for calls which do not conform to UUIDv7
2. This heuristic does not account for nested conditions (or the future of adding a less-than op)